### PR TITLE
Regular fb updates

### PR DESCRIPTION
**Pull request by davidhao3300**
Addresses #29 
This took way too long.

### CODE REVIEW

File: cal/schema.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 from flask.ext.sqlalchemy import SQLAlchemy
 import flask.ext.whooshalchemy as whooshalchemy
+from pytz import timezone
```
Comment (alanhdu) [inline cal/schema.py:3]: What's this import for?

File: cal/__init__.py [hunk 2]
```diff
@@ -3,13 +3,45 @@
...
+    celery = Celery(app.import_name, broker=app.config['CELERY_BROKER_URL'])
+    celery.conf.update(app.config)
+    TaskBase = celery.Task
+    class ContextTask(TaskBase):
+        abstract = True
+        def __call__(self, *args, **kwargs):
+            with app.app_context():
+                return TaskBase.__call__(self, *args, **kwargs)
+    celery.Task = ContextTask
+    return celery
```
Comment (alanhdu) [inline cal/__init__.py]: This seems overly complicated to just run functions within the `app.app_context()`. Maybe we should just manually specify `with app.app_context()` in `fb_task_test`?

Of course, that might get tiresome after awhile. Maybe this is the best solution...

File: cal/__init__.py [hunk 3]
```diff
@@ -3,13 +3,45 @@
...
+def create_app():
+    # Create the Flask app.
+    app = Flask(__name__)
+    app.config.from_object('config')
+    db.init_app(app)
+    whooshalchemy.whoosh_index(app, Event)
+    return app
+
+app = create_app()
+celery = make_celery(app)
```
Comment (alanhdu) [inline cal/__init__.py]: Is there a reason we're making `create_app` and `make_celery` functions? AFAIK, we only call these functions once, so I think it's kind of silly.

File: cal/__init__.py [hunk 4]
```diff
@@ -3,13 +3,45 @@
...
+
+# Background task
+@celery.task(name="cal.fb_task")
+def fb_task_test():
+    print "Starting FB update"
+    update_fb_events()
+    print "Finished FB update"
+    # Update every 30 minutes
+    fb_task_test.apply_async(countdown=1800)
```
Comment (alanhdu) [inline cal/__init__.py:29]: Maybe we should consider "printing" this to a log file (w/ time identifiers)? I'd rather not print anything to `stdout` unless an error occurs.

Also, should we create `task.py` specifically for celery tasks?

File: cal/__init__.py [hunk 5]
```diff
@@ -44,6 +76,7 @@ def home():
     return render_template('index.html', events=events)
 
 
+# TO REMOVE: Manual update route
```
Comment (alanhdu) [inline cal/__init__.py:69]: Should we just remove this route then?

File: config/__init__.py [hunk 6]
```diff
@@ -20,6 +20,7 @@
 
     BASEDIR = path.abspath(path.join(path.dirname(__file__), pardir))
     SQLALCHEMY_DATABASE_URI = "sqlite:///" + path.join(BASEDIR, "events.db")
+    CELERY_BROKER_URL = 'sqlalchemy+' + SQLALCHEMY_DATABASE_URI
```
Comment (alanhdu) [inline config/__init__.py:23]: Is there a reason for choosing this for `CELERY_BROKER_URL?`

File: cal/__init__.py (hunk 4 above)
Comment (davidhao3300) [inline cal/__init__.py:29]: Right this was mainly for debugging, when I couldn't get the task to start properly. I'll remove them for now.

File: cal/__init__.py (hunk 5 above)
Comment (davidhao3300) [inline cal/__init__.py:69]: I'm going to do a cleanup after this.

File: config/__init__.py (hunk 6 above)
Comment (davidhao3300) [inline config/__init__.py:23]: It's somewhat specified by the tutorial by Celery, I just followed their example on this.

File: cal/schema.py (hunk 1 above)
Comment (davidhao3300) [inline cal/schema.py:3]: Right now, the database can't handle dealing with naive/aware datetimes, so I have to manually strip and re-add the timezones. I'm going to look into alternatives after this.

**Comment by alanhdu** [comment]
LGTM

**Pull request merged**